testsuite/MDAnalysisTests/analysis/test_nucleicacids.py

```diff
@@ -0,0 +1,63 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+import MDAnalysis as mda
+import pytest
+from MDAnalysis.analysis.nucleicacids import WCDist, MinorDist, MajorDist, BaseSelect
+from MDAnalysisTests.datafiles import RNA_PSF, RNA_PDB
+from numpy.testing import assert_allclose
+
+
+@pytest.fixture(scope='module')
+def u():
+    return mda.Universe(RNA_PSF, RNA_PDB)
+
+
+def test_wc_dist(u):
+    sel = [(BaseSelect('RNAA', 1), BaseSelect('RNAA', 2)),
+           (BaseSelect('RNAA', 22), BaseSelect('RNAA', 23))]
+    WC = WCDist(u, sel)
```

Comment:
Is the user supposed to build selections in this way? That looks quite cumbersome. 

I think I'd prefer something where the user has to provide AtomGroups, e.g. for strand_1 and and strand_2 and the residues are matched up with zip(strand_1.residues, strand_2.residues). We can then provide a helper function to build these groups but the classes themselves don't have to much guessing baggage, except identifying the correct atoms for the distance.

Reply:
I'm trying to get an idea of how you want this to work, so the user provides the strands, then the residue pairs are formed in a list of tuples. The classes would then take the residue selections, use the zipped list to find the specific residues than grab the specific atoms. The super class would just accept a list of AtomGroups then merge them into a single AtomGroup. I think to do this I'd write a helper function to select the strands of DNA from a Universe. Is this what you are envisioning?

The question I have it are how do you select a specific strand of strand of DNA?

Reply:
I mostly used the named tuple for the sake of organizing inputs.

Reply:
>  so the user provides the strands, 

Yes, right now make it the user's responsibility to identify the WC pairs. `strand_1.residues[i]` will be considered base pairing with `strand_2.residues[i]`.

This is similar to the existing function, which asks the user to identify a single base pair via resid and segid. However, for your module we should move to an object-oriented approach and just work with AtomGroups instead of resid/segid selections.

> then the residue pairs are formed in a list of tuples. 

Yes.

> The classes would then take the residue selections, 

They take `strand_1.residues` and `strand_2.residues`

> use the zipped list to find the specific residues 

Yes.

> than grab the specific atoms. 

This is where the class does some actual work in order to identify the specific atoms that are required for the W-C base pair. Depending on the residue type, a different atom is chosen.

> The super class would just accept a list of AtomGroups then merge them into a single AtomGroup. 

I don't think that it would merge them into a single group. Ultimately you want to be able to use `calc_bonds()` on two arrays of positions. Each strand should have N residues and in the end, for each strand you should end up with an AtomGroup with N atoms, where the atom is either an N3 or and N1, depending on the nucleobase.

Reply:
I was thinking it would merge them into two atom groups which then are put into calc_bonds